x-pack/solutions/security/plugins/elastic_assistant/server/lib/langchain/graphs/constants.ts

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const NodeType = {
+  GENERATE_NODE: 'generate',
+  REFINE_NODE: 'refine',
+  RETRIEVE_ANONYMIZED_DOCS_NODE: 'retrieve_anonymized_docs',
+} as const;
```

Comment:
`NodeType` moved here from `x-pack/solutions/security/plugins/elastic_assistant/server/lib/langchain/graphs/index.ts` because that file imports `getDefaultAttackDiscoveryGraph`, and in 
In `default_attack_discovery_graph/index.ts`, it was importing `NodeType` from `../../../langchain/graphs`, which created a circular depdency, resulting in a runtime error when running AD evals.

Reply:
(via commit https://github.com/elastic/kibana/pull/216757/commits/48ded9315a027bc5a7eed2a1eb07741cb976af98)